org.eclipse.m2e.core/src/org/eclipse/m2e/core/embedder/IMaven.java

```diff
@@ -274,4 +273,18 @@ default <V> V execute(ICallable<V> callable, IProgressMonitor monitor) throws Co
    */
   <T> T lookup(Class<T> clazz) throws CoreException;
 
+
+  /**
+   * Creates and returns new, possibly nested, maven execution context for this Maven embedder.
+   * <p>
+   * <b>IMPORTANT:</b> When in the context of a particular project, it's usually better to use
+   * {@link IMavenProjectFacade#createExecutionContext()} which will include more project-specific configuration and
+   * will lead to more accurate and consistent results compared to Maven CLI commands.
+   * </p>
+   *
+   * @since 1.4
+   * @see IMavenProjectFacade#createExecutionContext()
+   */
+  IMavenExecutionContext createExecutionContext();
```

Comment:
And now instead of implement / define that twice you could add this (and default implement it) in IMavenExecutionContextFactory ;-)